homeassistant/components/frigidaire/climate.py

```diff
@@ -0,0 +1,270 @@
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up frigidaire from a config entry."""
+    client = hass.data[DOMAIN][entry.entry_id]
+
+    def get_entities(username: str, password: str) -> frigidaire.Frigidaire:
+        return client.get_appliances()
+
+    appliances = await hass.async_add_executor_job(
+        get_entities, entry.data["username"], entry.data["password"]
+    )
+
+    async_add_entities(
+        [FrigidaireClimate(client, appliance) for appliance in appliances],
+        update_before_add=True,
+    )
+
+
+class FrigidaireClimate(ClimateEntity):
+    """Representation of a Frigidaire appliance."""
+
+    def __init__(self, client, appliance):
+        """Build FrigidaireClimate.
+
+        client: the client used to contact the frigidaire API
+        appliance: the basic information about the frigidaire appliance, used to contact the API
+        """
+
+        self._client = client
+        self._appliance = appliance
+        self._details = None
+
+        # Entity Class Attributes
+        self._attr_unique_id = appliance.appliance_id
+        self._attr_name = appliance.nickname
+        self._attr_supported_features = SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+        self._attr_target_temperature_step = 1
+
+        # Although we can access the Frigidaire API to get updates, they are
+        # not reflected immediately after making a request. To improve the UX
+        # around this, we set assume_state to True
+        self._attr_assumed_state = True
+
+        self._attr_fan_modes = [
+            FAN_AUTO,
+            FAN_LOW,
+            FAN_MEDIUM,
+            FAN_HIGH,
+        ]
+
+        self._attr_hvac_modes = [
+            HVAC_MODE_OFF,
+            HVAC_MODE_COOL,
+            HVAC_MODE_AUTO,
+            HVAC_MODE_FAN_ONLY,
+        ]
+
+    @property
+    def assumed_state(self):
+        """Return True if unable to access real state of the entity."""
+        return self._attr_assumed_state
+
+    @property
+    def unique_id(self):
+        """Return unique ID based on Frigidaire ID."""
+        return self._attr_unique_id
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return self._attr_name
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return self._attr_supported_features
+
+    @property
+    def hvac_modes(self):
+        """List of available operation modes."""
+        return self._attr_hvac_modes
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return self._attr_target_temperature_step
+
+    @property
+    def fan_modes(self):
+        """List of available fan modes."""
+        return self._attr_fan_modes
```

Comment:
These are all implemented in the parent entity climate class :)

```suggestion
```

Reply:
This review comment was marked as resolved, however, nothing changed. Therefore unresolved it 😄